### DISCUSSION TITLE
Ability to animate slide changes only when dragging

### DISCUSSION THREAD

**Discussion by TomOne** [igordanchenko/yet-another-react-lightbox#76] [Ideas]
Currently, the `swipe` setting affects the animation on slide changes when dragging and also after clicking on the navigation buttons.

The default swipe animation after using the navigation buttons appears very “heavy” in my opinion and could distract some users when they browse through the slides, especially when they look at large screens which cover a large part of their field of view.

To avoid this and disable the animation, `swipe` could be set to `0`. But that leads to abrupt slide changes after dragging, which doesn’t result in a smooth experience either:

https://user-images.githubusercontent.com/4357973/220604958-55417553-a04c-4a52-8c63-216100388b69.mp4

Personally I’d be delighted if yet-another-react-lightbox (which is a great library otherwise) would support disabling the animation for navigation button clicks and keeping it when dragging from one slide to the next.

Would you be open to a PR for this?

**Comment by igordanchenko** [comment]
Thank you for reaching out and describing your concern. I can see why the current behavior may be undesirable. Do you have a suggestion on how to address this from the API (settings) point of view? PRs are always welcome here 👍

**Reply by TomOne** [reply]
Thanks, that sounds good.

How about a new `buttonNav?: number;` animation setting which allows to set the animation duration independently of `swipe`? Then `buttonNav: 0` would disable the animation when navigating with the buttons.

**Reply by igordanchenko** [reply]
I think it makes sense to treat nav buttons and keyboard navigation (left/right arrow keys) the same way in this case, so let's call the new prop `navigation`.

**Reply by igordanchenko** [reply]
I went ahead and released this feature in v2.5.0

**Reply by TomOne** [reply]
That’s great, thank you very much! Sorry it took me so long to reply.
